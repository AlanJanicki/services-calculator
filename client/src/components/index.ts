export { Card } from './Card/Card';
export { Checkbox } from './Checkbox/Checkbox';
export { Select } from './Select/Select';
export { Spinner } from './Spinner/Spinner';
