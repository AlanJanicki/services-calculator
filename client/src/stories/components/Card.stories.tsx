import { Card } from '../../components';

export default {
  component: Card,
  tags: ['autodocs'],
  title: 'Card'
};

export const Default = {
  args: {
    header: 'Kalkulator us≈Çug'
  }
};
