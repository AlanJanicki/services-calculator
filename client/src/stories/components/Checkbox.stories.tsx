import { Checkbox } from '../../components';

export default {
  component: Checkbox,
  tags: ['autodocs'],
  title: 'Checkbox'
};

export const Default = {
  args: {
    name: 'Internet'
  }
};
