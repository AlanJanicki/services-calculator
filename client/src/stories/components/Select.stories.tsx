import { Select } from '../../components';
import priceList from '../../mocks/servicesPriceList.json';

export default {
  component: Select,
  tags: ['autodocs'],
  title: 'Select'
};

export const Default = {
  args: {
    label: 'Wybierz rok',
    options: priceList.years
  }
};
