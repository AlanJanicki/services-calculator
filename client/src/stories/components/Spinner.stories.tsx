import { Spinner } from '../../components';

export default {
  component: Spinner,
  tags: ['autodocs'],
  title: 'Spinner'
};

export const Default = {};
