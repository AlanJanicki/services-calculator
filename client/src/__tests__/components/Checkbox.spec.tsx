import React from 'react';

import { Checkbox } from '../../components';
import { renderWithSnapshot } from '../common';

describe('<Checkbox/>', () => {
  it('matches snapshot', () =>
    renderWithSnapshot(<Checkbox checked={false} name='Internet' onChange={jest.fn} />));
});
