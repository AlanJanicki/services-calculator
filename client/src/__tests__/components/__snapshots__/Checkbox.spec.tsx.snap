// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: 2rem;
  margin-top: 0.3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 input,
.c0 label {
  cursor: pointer;
}

.c0 label {
  padding-left: 1rem;
}

@media (min-width:992px) {
  .c0 {
    margin-top: 0.5rem;
  }
}

@media (min-width:992px) {
  .c0 input {
    height: 1.2rem;
    aspect-ratio: 1;
  }
}

@media (min-width:992px) {
  .c0 label {
    font-size: 1.2rem;
  }
}

<div
    class="c0"
  >
    <input
      id="Internet"
      type="checkbox"
    />
    <label
      for="Internet"
    >
      Internet
    </label>
  </div>
</DocumentFragment>
`;
